client/src/encryption.rs

```diff
@@ -40,39 +45,85 @@ pub async fn decrypt<R: Rng + CryptoRng>(
         msg,
         from_address,
         &mut store.session_store,
-        &mut store.identity_store,
+        &mut store.identity_key_store,
         &mut store.pre_key_store,
-        &store.signed_pre_key_store,
+        &mut store.signed_pre_key_store,
         &mut store.kyber_pre_key_store,
         rng,
     )
     .await
 }
 
+#[derive(Debug)]
+pub enum PaddingError {
+    UnpadError,
+}
+
+impl Display for PaddingError {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        write!(f, "Could not unpad message - missing termination char 0x80")
+    }
+}
+
+impl Error for PaddingError {}
+
+fn get_padded_message_length(length: usize) -> usize {
+    let message_length_with_terminator = length + 1;
+    let mut message_part_count = message_length_with_terminator.div_euclid(160);
```

Comment:
This is just int division: https://doc.rust-lang.org/core/primitive.usize.html#method.div_euclid